ipa-core/src/protocol/hybrid/agg.rs

```diff
@@ -0,0 +1,217 @@
+            &reports[1].breakdown_key.to_bits(),
+        )
+        .await?;
+        let (value, _) = integer_add::<_, ThirtyTwoBitStep, 1>(
+            ctx.clone(),
+            record_id_v,
+            &reports[0].value.to_bits(),
+            &reports[1].value.to_bits(),
+        )
+        .await?;
+
+        agg_reports.push(AggregateableHybridReport::<BK, V> {
+            match_key: (),
+            breakdown_key: breakdown_key.collect_bits(),
+            value: value.collect_bits(),
+        });
+    }
+
+    Ok(agg_reports)
+}
+
+#[cfg(all(test, unit_test, feature = "in-memory-infra"))]
+pub mod test {
+    use super::aggregate_reports;
+    use crate::{
+        ff::boolean_array::{BA3, BA8},
+        report::hybrid::{
+            AggregateableHybridReport, IndistinguishableHybridReport, PrfHybridReport,
+        },
+        test_executor::run,
+        test_fixture::{hybrid::TestHybridRecord, Runner, TestWorld, TestWorldConfig, WithShards},
+    };
+
+    #[test]
+    fn aggregate_reports_test() {
+        run(|| async {
+            let records = vec![
+                TestHybridRecord::TestImpression {
+                    match_key: 12345,
+                    breakdown_key: 2,
+                },
+                TestHybridRecord::TestImpression {
+                    match_key: 23456,
+                    breakdown_key: 4,
+                },
+                TestHybridRecord::TestConversion {
+                    match_key: 23456,
+                    value: 1,
+                }, // attributed
+                TestHybridRecord::TestImpression {
+                    match_key: 34567,
+                    breakdown_key: 1,
+                },
+                TestHybridRecord::TestImpression {
+                    match_key: 45678,
+                    breakdown_key: 3,
+                },
+                TestHybridRecord::TestConversion {
+                    match_key: 45678,
+                    value: 2,
+                }, // attributed
+                TestHybridRecord::TestImpression {
+                    match_key: 56789,
+                    breakdown_key: 5,
+                },
+                TestHybridRecord::TestConversion {
+                    match_key: 67890,
+                    value: 3,
+                }, // NOT attributed
+                TestHybridRecord::TestImpression {
+                    match_key: 78901,
+                    breakdown_key: 2,
+                },
+                TestHybridRecord::TestConversion {
+                    match_key: 78901,
+                    value: 4,
+                }, // attributed twice
+                TestHybridRecord::TestConversion {
+                    match_key: 78901,
+                    value: 5,
+                }, // attributed twice
+                TestHybridRecord::TestImpression {
+                    match_key: 89012,
+                    breakdown_key: 4,
+                },
+                TestHybridRecord::TestConversion {
```

Comment:
what happens if we have two impressions and one conversion for the same match key?

Reply:
they should get filtered out, but that's a good test to include.